mitiq/executor/executor.py

```diff
@@ -147,6 +147,36 @@ def evaluate(
                 "Expected observable to be hermitian. Continue with caution."
             )
 
+        # Check executor and observable compatability with type hinting
+        # If FloatLike is specified as a return and observable is used
+        if self._executor_return_type in FloatLike and observable is not None:
+            # Type hinted as FloatLike and observable passed
+            if self._executor_return_type is not None:
+                raise ValueError(
+                    "When using a float like result, measurements should be "
+                    "included manually and an observable should not be "
+                    "used."
```

Comment:
```suggestion
                    "When using an executor which returns a float-like result, measurements should be added before the circuit is executed instead of with an observable."
```

WDYT? Just want to provide as much info as possible to ensure a user can fix the issue.

Reply:
I used your wording, just corrected the formatting in https://github.com/unitaryfund/mitiq/pull/2514/commits/19934a9d22d4ea381f6b7de7b113f5052b0a6ef8